templates/csharp/api.mustache

```diff
@@ -12,22 +12,50 @@ using Adyen.Service.Resource;
 
 namespace {{packageName}}.{{apiPackage}}
 {
+{{#operations}}
+    /// <summary>
+    /// {{classname}} Interface
+    /// </summary>
+    public interface I{{classname}}
+    {
+        {{#operation}}
+{{>method_documentation}}
+        {{#returnType}}
+        /// <returns><see cref="{{{returnType}}}"/>.</returns>
+        {{/returnType}}
+        {{#isDeprecated}}
+        [Obsolete]
+        {{/isDeprecated}}
+        {{{returnType}}}{{^returnType}}void{{/returnType}} {{#lambda.pascalcase}}{{vendorExtensions.x-methodName}}{{/lambda.pascalcase}}({{>api_parameters}});
+        
+        {{#supportsAsync}}
+{{>method_documentation}}
+        /// <param name="cancellationToken"> A CancellationToken enables cooperative cancellation between threads, thread pool work items, or Task objects.</param>{{#returnType}}
+        /// <returns>Task of <see cref="{{{returnType}}}{{^returnType}}void{{/returnType}}"/>.</returns>{{/returnType}}
+        {{#isDeprecated}}
+        [Obsolete]
+        {{/isDeprecated}}
+        {{#returnType}}Task<{{{.}}}>{{/returnType}}{{^returnType}}public async Task{{/returnType}} {{#lambda.pascalcase}}{{vendorExtensions.x-methodName}}{{/lambda.pascalcase}}Async({{>api_parameters_async}});
+        
+        {{/supportsAsync}}
+        {{/operation}}
+    }
+    {{/operations}}
+    
     {{#operations}}
     /// <summary>
     /// Represents a collection of functions to interact with the API endpoints
     /// </summary>
-    {{>visibility}} class {{classname}} : AbstractService
+    {{>visibility}} class {{classname}} : AbstractService, I{{classname}}
```

Comment:
@jillingk One last thing, please update the readme accordingly, there might be some services that are instantiated differently (also the names e.g. 'Checkout' in readme are incorrect) 👍

Reply:
@jillingk

Reply:
@Kwok-he-Chu, the Readme should be correct. We opted to define the directory for services with subservices (hence Checkout/ ) and if not the actual class name (eg StoredValueService). Old services that we didn't change, such as classic platforms and terminal API, I left as is. Does that makes sense to you, or would you rather see something else?

Reply:
I should've been more specific, sorry - let me clarify:

In https://github.com/Adyen/adyen-dotnet-api-library/blob/develop/README.md?plain=1#L48-L73 we create `new Checkout` which no longer exists and is pointed to IPaymentsService (developers should now create a new PaymentsService object instead) or even better: inject the services.AddSingleton<IPaymentsService, PaymentsService>() into their servicecollection container.

seealso https://github.com/Adyen/adyen-dotnet-api-library/blob/develop/Adyen/Service/Checkout/PaymentsService.cs
Note: The namespaces/references in the readme probably are incorrectly mentioned here


In https://github.com/Adyen/adyen-dotnet-api-library/blob/10.0.0-beta/README.md?plain=1#L11-L26, we should update the Service names (maybe even reference to have it link to the relative foldername in the readme, e.g. /checkout would go to the following folder: https://github.com/Adyen/adyen-dotnet-api-library/tree/10.0.0-beta/Adyen/Service/Checkout (You can have relative links by adding it as `[test](Adyen/Service/Checkout)` for example)

Reply:
@Kwok-he-Chu should we also update the README examples?

Reply:
@Kwok-he-Chu, great feedback thanks. I changed it except for the IServiceCollection injection. I want to create a little codesnippet for a generic .NET webapp setup but for that I first want to play around with it a bit so I know what works ;)